zarr/core.py

```diff
@@ -1930,76 +1923,44 @@ def _process_chunk(
         # store selected data in output
         out[out_selection] = tmp
 
-    def _chunk_getitem(self, chunk_coords, chunk_selection, out, out_selection,
```

Comment:
Is this so private that no one could have made use of it in a subclass?

Reply:
Yes, AFAICT.

Reply:
Yeah that's my understanding as well. Or at least by prefixing with `_` we have warned users this is an implementation detail (subject to change) that they shouldn't rely on.

Reply:
@jakirkham: this pre-dates me, so I defer. But if it's not documented somewhere we might want to review if that holds across the board. For me, `_x` is typically valid for subclassing by developers, otherwise it would be a `__x`. (i.e. public, protected, private in Java-parlance)

Reply:
It's [a Python thing generally]( https://stackoverflow.com/a/41761857 ) (not specific to Zarr).

Reply:
Single underscore if by far the most common thing to do, to show intent rather than enforce any privateness (which double underscore doesn't do either).

Reply:
@jakirkham: see also though https://stackoverflow.com/questions/41761645/when-to-use-one-or-two-underscore-in-python/41761857#comment99492184_41762806